### ISSUE TITLE
Granular right's (everything but full) denies agents to set themselfs as owner

### ISSUE THREAD

**Issue by MrGeneration** [zammad/zammad#2328]
### Infos:

* Used Zammad version: 2.7.x
* Installation method (source, package, ..): n/a
* Operating system: n/a
* Database + version: n/a
* Elasticsearch version: n/a
* Browser + version: Google Chrome 70

### Expected behavior:

* When setting role (or direct user) rights to "Read, Create, Change, Overview" agents can set the agent having this rights for a group as ticket owner without hassle.

### Actual behavior:

* When setting role (or direct user) rights to "Read, Create, Change, Overview", no agent is able to set the agent having this rights as Ticket-Owner, even though this gives him full rights to the ticket.
When setting role (or direct user) rights to "Full", agents are able to set the agent with those rights as ticket owner (but only for those group he has "FULL"-Right).

### Steps to reproduce the behavior:

* Create atleast two groups (ensure no existing role rights are impacting them with rights)
* Edit a role or user the rights "Read, Create, Change, Overview" for both test groups (if you use a role for that, ensure the testagent is assigned to this role)
* Take over control over that user (relogin to testuser is ok too)
* try creating a ticket inside one of your test groups while assigning your agent as owner (won't work)
  Same applies for existing tickets or other agents trying to assign the test agent as owner in the test group
* No go back to role (or user) right-settings and change one of the two entries to "FULL" (remove the other ticks for that group)
* Take back control over the test agent and try again
  The difference: As soon as you try to assign the agent as owner to the group he has FULL-right on, you can do that

Screencast showing that issue:

![granular-role-rights](https://user-images.githubusercontent.com/6549061/47937266-73b3bd80-dee0-11e8-8161-7d8b05278862.gif)

Yes I'm sure this is a bug and no feature request or a general question.

This impacts Ticket#1029168

**Comment by SummerStorm** [comment]
This is actually a feature working as designed, though it has not been well documented so far. I have updated the documentation [here](https://zammad-admin-documentation.readthedocs.io/en/latest/manage-roles.html#agent).

Basically any role with the Full permission is considered an agent, and thus can be assigned as a ticket owner. Conversely, any role without the Full permission, even though they have all other permissions, cannot be assigned as a ticket owner (whether by themselves or someone else). 

The use case of a role having every permission except for Full is the supervisor role, where they can oversee other agents, but should not get assigned any tickets themselves.

![permission-example-use-case-supervisor](https://user-images.githubusercontent.com/7030950/48467256-52799800-e823-11e8-86c2-3b23e2562213.jpg)

In other words, the Full permission is equivalent to the combination of 5 separate permissions: Read, Create, Change, Overview, Own_Tickets. Without the Full permission you cannot own tickets within that group.
